File: laundry/wash_wallet_be/app/Models/CustomerAccount.php
<?php

declare(strict_types=1);

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Table;

use App\Models\CustomerAddress;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\HasOne;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;
use Laravel\Sanctum\HasApiTokens;
use Illuminate\Foundation\Auth\User as Authenticatable;

#[Table('customer_accounts')]
class CustomerAccount extends Authenticatable
{
    use HasApiTokens, HasFactory, SoftDeletes;

    protected $fillable = [
        'phone',
        'name',
        'email',
        'gender',
        'password',
        'date_of_birth',
        'avatar',
        'is_verified',
        'is_active',
        'last_login_at',
        'fcm_token',
        'deposit_balance',
    ];

    protected $hidden = [
        'password',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'password' => 'hashed',
            'date_of_birth' => 'date',
            'is_verified' => 'boolean',
            'is_active' => 'boolean',
            'last_login_at' => 'datetime',
            'deposit_balance' => 'integer',
        ];
    }

    public function customers(): HasMany
    {
        return $this->hasMany(Customer::class);
    }

    public function customerTopups(): HasMany
    {
        return $this->hasMany(CustomerTopup::class);
    }

    public function orders(): HasMany
    {
        return $this->hasMany(Order::class);
    }

    public function addresses(): HasMany
    {
        return $this->hasMany(CustomerAddress::class);
    }

    public function primaryAddress(): HasOne
    {
        return $this->hasOne(CustomerAddress::class)->where('is_primary', true);
    }

    public function scopeByPhone($query, $phone)
    {
        return $query->where('phone', $phone);
    }
}

File: laundry/wash_wallet/packages/wash_wallet_domain/lib/src/entities/customer_account.dart
import 'package:equatable/equatable.dart';

class CustomerAccount extends Equatable {
  final int id;
  final String phone;
  final String name;
  final String? email;
  final String? gender;
  final String? avatar;
  final String? dateOfBirth;
  final bool isVerified;
  final bool isActive;
  final String? lastLoginAt;
  final String? fcmToken;
  final int depositBalance;
  final bool? hasPassword;

  const CustomerAccount({
    required this.id,
    required this.phone,
    required this.name,
    this.email,
    this.gender,
    this.avatar,
    this.dateOfBirth,
    required this.isVerified,
    required this.isActive,
    this.lastLoginAt,
    this.fcmToken,
    this.depositBalance = 0,
    this.hasPassword,
  });

  @override
  List<Object?> get props => [
    id,
    phone,
    name,
    email,
    gender,
    avatar,
    dateOfBirth,
    isVerified,
    isActive,
    lastLoginAt,
    fcmToken,
    depositBalance,
    hasPassword,
  ];
}
